src/DataFramesMeta.jl

```diff
@@ -519,9 +511,20 @@ end
 ##############################################################################
 
 function based_on_helper(x, args...)
-    with_args =
-        with_anonymous(:($DataFrame($(map(replace_equals_with_kw, args)...))))
-    :(DataFrames.combine($with_args, $x))
+    if length(args) == 1 &&
```

Comment:
We only allow one argument when people want to output a table object, similar to `combine`. In particular, with current `combine` the table output needs to go in the first argument, and we can't ensure that if there is more than one argument.

Reply:
OK, then I'd say it's worth adding a comment to explain this. Also `combinefun` may not be the best name for this, as `combine` accepts several forms. What's special AFAICT is that there's no output name provided, right?

Reply:
Yeah what's special is that we don't have `y = f(:x)` but still want `:x` to represent the vector `df.x`. So we want to apply `replace_syms!` but not create a new column the normal way. 

This will definitely get re-worked in the future, but probably in a breaking way with some new syntax. This flag is just for the non-breaking-ness of the PR. 

I think I will rename it to `combinetable`.

Reply:
OK I think I still don't really get what happens. :-p